src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,91 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final double DEFAULT_LOAD_FACTOR = 0.75;
+    private double threshold;
+    private Node<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        table = new Node[INITIAL_CAPACITY];
+        threshold = INITIAL_CAPACITY * DEFAULT_LOAD_FACTOR;
+    }
 
     @Override
     public void put(K key, V value) {
+        int index = hash(key);
+        Node<K, V> newNode = new Node<>(key, value);
+        Node<K, V> node = table[index];
+        while (node != null) {
+            if (Objects.equals(node.key, newNode.key)) {
+                node.value = newNode.value;
+                return;
+            }
+            node = node.next;
+        }
+        newNode.next = table[index];
+        table[index] = newNode;
+        size++;
 
+        if (size >= threshold) {
+            resize();
+        }
     }
 
     @Override
     public V getValue(K key) {
+        int index = hash(key);
+        Node<K, V> node = table[index];
+        while (node != null) {
+            if (Objects.equals(node.key, key)) {
+                return node.value;
+            }
+            node = node.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private boolean isEmpty() {
+        return size == 0;
+    }
+
+    private int hash(K key) {
+        return key == null ? 0 : Math.abs(key.hashCode() % table.length);
+    }
+
+    private void resize() {
+        int newCapacity = table.length << 1;
+        final Node<K, V>[] oldTable = table;
+        table = new Node[newCapacity];
+        size = 0;
+        threshold = newCapacity * DEFAULT_LOAD_FACTOR;
+
+        for (Node<K, V> node : oldTable) {
+            while (node != null) {
+                put(node.key, node.value);
+                node = node.next;
+            }
+        }
+    }
+
+    static class Node<K, V> {
+        private final int hash;
+        private final K key;
+        private V value;
+        private Node<K, V> next;
+
+        public Node(K key, V value) {
```

Comment:
Since the Node class is defined as a private static class within the MyHashMap class, it is only accessible within the MyHashMap class, regardless of the visibility of its constructor. In this case, the visibility of the Node constructor (public, private, or default) won't make any difference from outside of the MyHashMap class, since the Node class itself is private to MyHashMap.

However, from the perspective of code organization and readability, it is a good practice to make the constructor of the Node class private, as it should only be used within the MyHashMap class itself. It makes it clear that the Node class is solely used as an internal data structure of the MyHashMap class and should not be directly instantiated elsewhere in the MyHashMap class:
```suggestion
        private Node(K key, V value) {
```